fireworks/core/firework.py

```diff
@@ -393,7 +393,7 @@ class Launch(FWSerializable, object):
     """
 
     def __init__(self, state, launch_dir, fworker=None, host=None, ip=None, trackers=None,
-                 action=None, state_history=None, launch_id=None, fw_id=None):
+                 action=None, action_gridfs_id=None, state_history=None, launch_id=None, fw_id=None):
```

Comment:
I would prefer it if the core Launch object itself did not have an action_gridfs_id attribute. It's better if the Launch object can not contain too many details of the specific database implementation that will be used to store it.

Is there a way to structure this so that the existing "action" attribute can easily distinguish whether it's a normal action or a gridfs id? e.g., perhaps if the action looks like this: action={"gridfs_id": "adsfasdf"} then the LaunchPad will know that the action needs retrieval from GridFS.

Reply:
My point in adding one more attribute was to avoid potential problems while deserializing the Launch object (now the dictionary in the db may not contain an FWAction), but I see your point in wanting to preserve the Launch interface.  
I have implemented the modification that you suggested. As long as the Launch is generated or retrieved using theLaunchPad it should be safe.